fix: PK fix for conformance_pack_rule_compliances

spangenberg opened the pull request:
The PK was set on the parent but should be on default `cq_id`.
Also the filter for the Details was not applied.

```
cloudquery $ cloudquery fetch aws:config.conformance_packs
An update to CloudQuery core is available: 0.22.10!

Initializing CloudQuery Providers...

Finished provider initialization...

Upgrading CloudQuery providers aws

✓ Upgraded provider aws to latest successfully.

Finished upgrading providers...

Starting provider fetch...

✓ cq-provider-aws@latest fetch complete    3s   Finished Resources: 1/1

Fetch Diagnostics for provider aws:

Resource: config.conformance_packs Type: Database Severity: Error
	Summary: failed to insert to table "aws_config_conformance_pack_rule_compliances": ERROR: duplicate key value violates unique constraint "aws_config_conformance_pack_rule_compliances_pk" (SQLSTATE 23505)
	Detail: duplicate key value violates unique constraint "aws_config_conformance_pack_rule_compliances_pk". [Repeated:36]

Provider fetch complete.

Provider aws fetch summary: ✓ Total Resources fetched: 1	 ⚠️ Warnings: 0	 ❌ Errors: 1
```

With fix:

```
cloudquery $ cloudquery fetch aws:config.conformance_packs
An update to CloudQuery core is available: 0.22.10!

Initializing CloudQuery Providers...

Finished provider initialization...

Upgrading CloudQuery providers aws

✓ Upgraded provider aws to latest successfully.

Finished upgrading providers...

Starting provider fetch...

✓ cq-provider-aws@latest fetch complete    3s   Finished Resources: 1/1

Provider fetch complete.

Provider aws fetch summary: ✓ Total Resources fetched: 1	 ⚠️ Warnings: 0	 ❌ Errors: 0
cloudquery $
```

roneli reviewed (changes requested):
There seems to be migration errors, based on passed experience would also verify after test fixes.
Fetch previous version -> upgrade to this version with data. 
Downgrade back to previous version. 
Run on clean db

spangenberg commented:
> There seems to be migration errors, based on passed experience would also verify after test fixes. Fetch previous version -> upgrade to this version with data. Downgrade back to previous version. Run on clean db

It's fixed now, tested everything as well by hand with existing data.

roneli commented on resources/provider/migrations/timescale/28_v0.11.2.down.sql:
```diff
@@ -1,3 +1,11 @@
 
 ALTER TABLE IF EXISTS "aws_sns_topics" DROP COLUMN IF EXISTS "tags";
+
+DROP INDEX "aws_config_conformance_pack_rule_compli_cq_fetch_date_cq_id_idx";
```
DROP IF EXISTS

roneli replied:
same with normal

roneli commented on resources/provider/migrations/timescale/28_v0.11.2.down.sql:
```diff
@@ -1,3 +1,11 @@
+DROP INDEX "aws_config_conformance_pack_rule_compli_cq_fetch_date_cq_id_idx";
+
+ALTER TABLE "aws_config_conformance_pack_rule_compliances"
+  DROP CONSTRAINT aws_config_conformance_pack_rule_compliances_pk,
```
IF EXISTS

roneli reviewed (approved):
LGTM

bbernays commented on resources/provider/migrations/postgres/28_v0.11.2.down.sql:
```diff
@@ -1,3 +1,7 @@
 
-ALTER TABLE IF EXISTS "aws_sns_topics" DROP COLUMN IF EXISTS "tags";
\ No newline at end of file
+ALTER TABLE IF EXISTS "aws_sns_topics" DROP COLUMN IF EXISTS "tags";
+
+ALTER TABLE IF EXISTS "aws_config_conformance_pack_rule_compliances"
+  DROP CONSTRAINT aws_config_conformance_pack_rule_compliances_pk,
+  ADD CONSTRAINT aws_config_conformance_pack_rule_compliances_pk PRIMARY KEY (conformance_pack_cq_id);
```
The version needs to be updated to 0.11.3

spangenberg committed:
fix: PK fix for conformance_pack_rule_compliances

The PK was set on the parent but should be on default `cq_id`.
Also the filter for the Details was not applied.

```
cloudquery $ cloudquery fetch aws:config.conformance_packs
An update to CloudQuery core is available: 0.22.10!

Initializing CloudQuery Providers...

Finished provider initialization...

Upgrading CloudQuery providers aws

✓ Upgraded provider aws to latest successfully.

Finished upgrading providers...

Starting provider fetch...

✓ cq-provider-aws@latest fetch complete    3s   Finished Resources: 1/1

Fetch Diagnostics for provider aws:

Resource: config.conformance_packs Type: Database Severity: Error
	Summary: failed to insert to table "aws_config_conformance_pack_rule_compliances": ERROR: duplicate key value violates unique constraint "aws_config_conformance_pack_rule_compliances_pk" (SQLSTATE 23505)
	Detail: duplicate key value violates unique constraint "aws_config_conformance_pack_rule_compliances_pk". [Repeated:36]

Provider fetch complete.

Provider aws fetch summary: ✓ Total Resources fetched: 1	 ⚠️ Warnings: 0	 ❌ Errors: 1
```

With fix:

```
cloudquery $ cloudquery fetch aws:config.conformance_packs
An update to CloudQuery core is available: 0.22.10!

Initializing CloudQuery Providers...

Finished provider initialization...

Upgrading CloudQuery providers aws

✓ Upgraded provider aws to latest successfully.

Finished upgrading providers...

Starting provider fetch...

✓ cq-provider-aws@latest fetch complete    3s   Finished Resources: 1/1

Provider fetch complete.

Provider aws fetch summary: ✓ Total Resources fetched: 1	 ⚠️ Warnings: 0	 ❌ Errors: 0
cloudquery $
```
